### PR TITLE
[WFLY-19681] minimize lock hold while iterating across existing entries in refresh task

### PR DESCRIPTION
**Pull request by aogburn**
https://issues.redhat.com/browse/WFLY-19681

### CODE REVIEW

**Comment by bstansberry** [comment]
@aogburn Can you correct the commit message? It's the wrong JIRA.

**Comment by aogburn** [comment]
Yes, changed! Sorry, I moved the original jira instead of my cloned one and forgot to fix up the commit message.

File: ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/database/DatabaseTimerPersistence.java [hunk 1]
```diff
@@ -1099,11 +1099,11 @@ public void run() {
                                 }
                             }
 
-                            synchronized (DatabaseTimerPersistence.this) {
-                                Set<String> timers = knownTimerIds.get(timedObjectId);
-                                for (String timer : existing) {
-                                    TimerImpl timer1 = timerService.getTimer(timer);
-                                    if (timer1 != null && timer1.getState() != TimerState.CREATED) {
+                            Set<String> timers = knownTimerIds.get(timedObjectId);
```
Comment (bstansberry) [inline ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/database/DatabaseTimerPersistence.java:1102]: All other access to knownTimerIds in this class is guarded by 'synchronized (DatabaseTimerPersistence.this)' (or synchronized (this) in DatabaseTimerPersistence itself.

Except the first lie of loadActiveTimers, which is probably a bug.

Changing the concurrency handling for just one call seems risky.

The sets in the map also aren't thread safe.

Comment (aogburn) [reply ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/database/DatabaseTimerPersistence.java]: D'oh, thanks for catching that Brian.  Added the knownTImerIds back within a sync there.

Comment (aogburn) [reply ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/database/DatabaseTimerPersistence.java]: Any further thoughts on this @bstansberry @tadamski ?

**Comment by bstansberry** [comment]
Thanks @aogburn

**Pull request merged**